common/base_c/defs.h

```diff
@@ -51,7 +74,11 @@
 #define M_PI 3.14159265358979323846
 #endif
 
-#define NELEM(arr) ((int)(sizeof(arr) / sizeof(arr[0])))
+#ifdef __cplusplus
```

Comment:
I would avoid adding C++ code to `base_c/defs.h`, since it's explicitly named as a C header.